westcoast/src/main/java/frc/commands/ActuateTwoDoubleSolenoids.java

```diff
@@ -0,0 +1,107 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.commands;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+
+public class ActuateTwoDoubleSolenoids extends Command {
+  private final DoubleSolenoid solenoid;
+  private final DoubleSolenoid solenoid2;
+	private final Value direction;
+  private final Value endDirection;
+  private int v = 0;
```

Comment:
`v` tells me nothing.

What this state is supposed to represent? Naming is important for code legibility. Usually you can get more compact as your scope gets deeper.

Reply:
Also the naked `ints`. `0`, `1`, `2`, and `3` give names to make theirs semantic clear, if nothing else it will help keep the code maintainable as changing the value in one place will change everywhere the name is uses.
```java
 static final int CONST_0 = 0;
```
 is how is done in Java.  Make sure to avoid use `CONST_0` if you can find a better name for it.

Anyway an `enum`  looks a better choice to use here. Check them out: [Enums in Java](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html) (but try to avoid overuse it)

Reply:
cleaned up, used enum took me way to long to figure out how to work them, still not quite understanding them, but I get the specific use I did.